flox/core.py

```diff
@@ -98,7 +102,7 @@ def _is_first_last_reduction(func: T_Agg) -> bool:
     return isinstance(func, str) and func in ["nanfirst", "nanlast", "first", "last"]
 
 
-def _get_expected_groups(by: T_By, sort: bool) -> pd.Index:
+def _get_expected_groups(by: T_By, sort: bool) -> T_ExpectIndex:
```

Comment:
Please leave this as `pd.Index`. https://matthewrocklin.com/avoid-indirection.html

Reply:
I disagree. The point is to indicate that it belongs to the expect-family of types and therefore treated equally.

I find xarray has big issues with this where the same parameter has different typing in other parts of the code. It feels inconsistent and leads to unnecessary bugs in my opinion. It's a reason why we're defining more and more types in  https://github.com/pydata/xarray/blob/main/xarray/core/types.py.

Renaming like this also shows how many different types there are for the expect-family and that maybe a refactor to normalize the types is in order.